### ISSUE TITLE
ExtractPolygonalPrismData & ConcaveHull yeild wrong results

### ISSUE THREAD

**Issue by yoavmil** [PointCloudLibrary/pcl#5163]
tl;dr 
If the concave hull is made of multiple polygons, when the prism checks if a point is inside, the result is sometimes wrong.
I also have the solution.

To test it, I made a cloud of 2 rings, found there concave hull, and applied `ExtractPolygonalPrismData` on them.

before fix:
![image](https://user-images.githubusercontent.com/5158716/153456360-c87b2234-c698-433e-88b4-8ee5be09b346.png)

after fix:
![image](https://user-images.githubusercontent.com/5158716/153456632-09b743b5-b7fc-4150-b97b-b3584a54e759.png)

**To Reproduce**
I've put a working code in [my github](https://github.com/yoavmil/pcl-sandbox)

pcl:x64-windows 1.12.0#4

**Possible Solution**
I've written and tested a solution.
But first I'll explain the bug. each point is tested if it is inside a polygon [using this code](www.visibone.com/inpoly).
![image](https://user-images.githubusercontent.com/5158716/153457527-303374fc-bb0e-4235-b3f0-bd3e23f313ec.png)
The problem is that if there is more than one ring in the polyongs, the code flattens them to one polygons, adding more segments that doesn't exist.

To solve this, I took the `polygons` from `ConcaveHull`, which discribe the rings and their order, and passed it to `ExtractPolygonalPrismData::segment`. Then, I added an overload to `isXYPointIn2DXYPolygon` that recieved the rings order as well, and check each of them seperatly. 

[The code is also in my repo.](https://github.com/yoavmil/pcl-sandbox/blob/main/modified_pcl_files/extract_polygonal_prism_data.h)

I could do a PR if someone would read and aprove.

**Comment by larshg** [comment]
Added in #5168
